core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -54,7 +54,7 @@ import org.apache.spark.util.{AccumulatorV2, ThreadUtils, Utils}
 private[spark] class TaskSchedulerImpl private[scheduler](
     val sc: SparkContext,
     val maxTaskFailures: Int,
-    private[scheduler] val blacklistTrackerOpt: Option[BlacklistTracker],
+    mockBlacklistTracker: Option[BlacklistTracker] = None,
```

Comment:
whats the point of this refactoring?  I think I see the point -- you are showing that these constructors are only directly called w/ a mock blacklist tracker, but overall I think it actually makes it harder to understand the code, its odd to see something called "mock" in the main code.